src/worktree.rs

```diff
@@ -0,0 +1,333 @@
+    pub fn name(&self) -> Option<&str> {
+        unsafe {
+            crate::opt_bytes(self, raw::git_worktree_name(self.raw))
+                .and_then(|s| str::from_utf8(s).ok())
+        }
+    }
+
+    /// Retrieves the path to the worktree
+    ///
+    /// This is the path to the top-level of the source and not the path to the
+    /// .git file within the worktree. This path can be passed to
+    /// repo::Repository::open.
+    pub fn path(&self) -> &Path {
+        unsafe {
+            util::bytes2path(crate::opt_bytes(self, raw::git_worktree_path(self.raw)).unwrap())
+        }
+    }
+
+    /// Validates the worktree
+    ///
+    /// This checks that it still exists on the
+    /// filesystem and that the metadata is correct
+    pub fn validate(&self) -> Result<(), Error> {
+        unsafe {
+            call::c_try(raw::git_worktree_validate(call::convert(&self.raw)))?;
+        }
+        Ok(())
+    }
+
+    /// Locks the worktree
+    pub fn lock(&self, reason: Option<&str>) -> Result<(), Error> {
+        let reason = crate::opt_cstr(reason)?;
+        unsafe {
+            try_call!(raw::git_worktree_lock(self.raw, reason));
+        }
+        Ok(())
+    }
+
+    /// Unlocks the worktree
+    pub fn unlock(&self) -> Result<(), Error> {
+        unsafe {
+            call::c_try(raw::git_worktree_unlock(call::convert(&self.raw)))?;
+        }
+        Ok(())
+    }
+
+    /// Checks if worktree is locked
+    pub fn is_locked(&self) -> Result<WorktreeLockStatus, Error> {
+        let buf = Buf::new();
+        unsafe {
+            match try_call!(raw::git_worktree_is_locked(buf.raw(), self.raw)) {
+                0 => Ok(WorktreeLockStatus::Unlocked),
+                _ => {
+                    let v = buf.to_vec();
+                    Ok(WorktreeLockStatus::Locked(match v.len() {
+                        0 => None,
+                        _ => String::from_utf8(v).ok(),
+                    }))
+                }
+            }
+        }
+    }
+
+    /// Prunes the worktree
+    pub fn prune(&self, opts: Option<&WorktreePruneOptions>) -> Result<(), Error> {
+        // When successful the worktree should be removed however the backing structure
+        // of the git_worktree should still be valid.
+        unsafe {
+            let mut opts = opts.map(|o| o.raw());
+            try_call!(raw::git_worktree_prune(self.raw, opts.as_mut()));
+        }
+        Ok(())
+    }
+
+    /// Checks if the worktree is prunable
+    pub fn is_prunable(&self, opts: Option<&WorktreePruneOptions>) -> Result<bool, Error> {
+        unsafe {
+            let mut opts = opts.map(|o| o.raw());
+            let rv = try_call!(raw::git_worktree_is_prunable(self.raw, opts.as_mut()));
+            Ok(rv != 0)
+        }
+    }
+
+    /// Opens the repository from the worktree
+    pub fn to_repository(&self) -> Result<Repository, Error> {
+        Repository::open_from_worktree(self)
```

Comment:
I personally prefer to stick close to the original `git2` API where possible, so could this and the other convenience alias be removed?